packages/test/test-gc-sweep-tests/src/test/gcSweepTest.spec.ts

```diff
@@ -0,0 +1,355 @@
+        "TestDataObject",
+        TestDataObject,
+        [],
+        []);
+
+    const runtimeOptions: IContainerRuntimeOptions = {
+        summaryOptions: {
+            summaryConfigOverrides: {
+                state: "disableHeuristics",
+                initialSummarizerDelayMs: 0,
+                summarizerClientElection: false,
+                maxAckWaitTime: 100,
+                maxOpsSinceLastSummary: 100,
+            },
+        },
+        gcOptions: {
+            gcAllowed: true,
+        },
+    };
+    const innerRequestHandler = async (request: IRequest, runtime: IContainerRuntimeBase) =>
+        runtime.IFluidHandleContext.resolveHandle(request);
+    const runtimeFactory = new ContainerRuntimeFactoryWithDefaultDataStore(
+        dataObjectFactory,
+        [
+            [dataObjectFactory.type, Promise.resolve(dataObjectFactory)],
+        ],
+        undefined,
+        [innerRequestHandler],
+        runtimeOptions,
+    );
+
+    const defaultSessionExpiryDurationMs = 3000; // 3 seconds
+
+    // Enable config provider setting to write GC data at the root.
+    const settings = {
+        "Fluid.GarbageCollection.RunSessionExpiry": "true",
+        "Fluid.GarbageCollection.TestOverride.SessionExpiryMs": defaultSessionExpiryDurationMs,
+    };
+    const configProvider = mockConfigProvider(settings);
+
+    const uploadedSummaries: ISummaryTree[] = [];
+
+    let overrideLogger: IgnoreErrorLogger;
+
+    let mainContainer: IContainer;
+    let connectedContainers: IContainer[];
+    let closedContainers: IContainer[];
+    let testNodes: Map<string, TestNode>;
+    let unrefDataStores: Set<string>;
+    let refDataStores: Set<string>;
+    let rootDataStores: Set<string>;
+
+    const createContainer = async (): Promise<IContainer> => {
+        return provider.createContainer(runtimeFactory, { configProvider });
+    };
+    const loadContainer = async (): Promise<IContainer> => {
+        return provider.loadContainer(runtimeFactory, { configProvider });
+    };
+
+    const sleep = async (ms: number) => new Promise((resolve) => setTimeout(resolve, ms));
+
+    beforeEach(async () => {
+        provider = getTestObjectProvider({
+            syncSummarizer: true,
+        });
+        overrideLogger = new IgnoreErrorLogger(provider.logger);
+        provider.setOverrideLogger(overrideLogger);
+
+        // Create a Container for the first client.
+        mainContainer = await createContainer();
+        const mainDataStore = await requestFluidObject<TestDataObject>(mainContainer, "default");
+        connectedContainers = [];
+        closedContainers = [];
+        testNodes = new Map();
+        testNodes.set(mainDataStore.id, new TestNode(mainDataStore.id));
+        unrefDataStores = new Set();
+        refDataStores = new Set();
+        rootDataStores = new Set();
+        refDataStores.add(mainDataStore.id);
+        rootDataStores.add(mainDataStore.id);
+
+        await provider.ensureSynchronized();
```

Comment:
What does this do for you practically?  Have you done anything yet that generates ops?  I guess createContainer probably does, but why do you need to wait for those acks?  Or maybe I'm out of touch with what ensureSynchronized actually does.

Reply:
It may not be needed, does creating a datastore generate an op?

Reply:
Yeah not sure... You're talking about the default dataStore?  Seems like it would, I guess.

Reply:
I'll try removing it